### PR TITLE
Add initial libraries for ssh support.

### PR DESCRIPTION
**Pull request by dlorenc**
I'll rebase this up after some more clean ups.

This doesn't hook anything up yet, but adds some functions and tests for the SSH signature format.

### CODE REVIEW

File: pkg/pki/ssh/sign.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	Signature     string
+}
+
+const (
+	magicHeader   = "SSHSIG"
+	hashAlgorithm = "sha512"
+)
+
+func sign(s ssh.AlgorithmSigner, m io.Reader) (*ssh.Signature, error) {
+	data, err := ioutil.ReadAll(m)
```
Comment (bobcallaway) [inline pkg/pki/ssh/sign.go]: I would do an `io.Copy` from `m` to `hf` rather than `ReadAll` and then `Write`.

File: pkg/pki/ssh/sign.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	dataMessageWrapper = append([]byte(magicHeader), dataMessageWrapper...)
+
+	sig, err := s.SignWithAlgorithm(rand.Reader, dataMessageWrapper, ssh.SigAlgoRSASHA2512)
+	if err != nil {
+		return nil, err
+	}
+	return sig, nil
+}
+
+func Sign(sshPrivateKey string, data []byte) (string, error) {
```
Comment (bobcallaway) [inline pkg/pki/ssh/sign.go]: can you make `data` an `io.Reader` and just pass it through to the `sign()` call?

File: pkg/pki/ssh/verify.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package ssh
+
+import (
+	"crypto/sha512"
+
+	"golang.org/x/crypto/ssh"
+)
+
+func Verify(message []byte, armoredSignature string, publicKey []byte) error {
```
Comment (bobcallaway) [inline pkg/pki/ssh/verify.go]: use `io.Reader` instead of `byte[]` in case it is a large payload

File: pkg/pki/ssh/verify.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		return err
+	}
+
+	desiredPk, _, _, _, err := ssh.ParseAuthorizedKey(publicKey)
+	if err != nil {
+		return err
+	}
+
+	// Hash the message so we can verify it against the signature.
+	h := sha512.New()
```
Comment (bobcallaway) [inline pkg/pki/ssh/verify.go:22]: `io.Copy` from reader straight to writer `&h`

File: pkg/pki/ssh/encode.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		Version:       1,
+		PublicKey:     string(p.Marshal()),
+		Namespace:     "file",
+		HashAlgorithm: hashAlgorithm,
+		Signature:     string(ssh.Marshal(s)),
+	}
+	copy(sig.MagicHeader[:], []byte(magicHeader))
+
+	enc := pem.EncodeToMemory(&pem.Block{
+		Type:  "SSH SIGNATURE",
```
Comment (bobcallaway) [inline pkg/pki/ssh/encode.go]: make this into `const`?

File: pkg/pki/ssh/sign.go (hunk 1 above)
Comment (dlorenc) [reply pkg/pki/ssh/sign.go]: Nice! done.

File: pkg/pki/ssh/verify.go (hunk 3 above)
Comment (dlorenc) [reply pkg/pki/ssh/verify.go]: Done.

File: pkg/pki/ssh/verify.go (hunk 4 above)
Comment (dlorenc) [reply pkg/pki/ssh/verify.go:22]: Done!

**Review by bobcallaway — APPROVED** [review]
lgtm

**Comment by dlorenc** [comment]
I'm going to squash this and then merge if tests still pass!

**Pull request merged**